src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -517,10 +574,59 @@ public int MultiDimensionalArrayRank
             }
         }
 
+        public bool IsValueType
+        {
+            get
+            {
+                return (Flags & enum_flag_Category_ValueType_Mask) == enum_flag_Category_ValueType;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public TypeHandle GetArrayElementTypeHandle()
+        {
+            return *(TypeHandle*)&((MethodTable*)Unsafe.AsPointer(ref this))->ElementType;
```

Comment:
I still insist it should be either `new TypeHandle(ElementType)` or `ElementType` field must be of `TypeHandle` type - the current piece of code is very hard to read 😐

Reply:
I agree that this is hard to read. @Sergio0694 Could you please do something about this?